sdk/nodejs/runtime/resource.ts

```diff
@@ -501,6 +503,38 @@ export function registerResourceOutputs(res: Resource, outputs: Inputs | Promise
     }, false);
 }
 
+function isAny(o: any): o is any {
+    return true;
+}
+
+/**
+ * listResourceOutputs returns the resource outputs (if any) for a stack, or an error if the stack
+ * cannot be found. Resources are retrieved from the latest stack snapshot, which may include
+ * ongoing updates.
+ *
+ * @param stackName Name of stack to retrieve resource outputs for. Defaults to the current stack.
+ * @param typeFilter A [type
+ * guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards)
+ * that specifies which resource types to list outputs of.
+ *
+ * @example pulumi.runtime.listResourceOutput(aws.s3.Bucket.isInstance)
+ */
+export function listResourceOutputs<U extends CustomResource = any>(
```

Comment:
I do wonder about the shape of this API.  The optional first parameter is kinda wonky, if we expect the common case to be running queries against the "current stack", which I feel like will be the common case?  I expect that in general, I'm going to have my Pulumi project and then one or more `.ts` files that have operational queries I'm interested in.  In that world, I'm going to run something like:

```
$ pulumi query findPublicBuckets.ts
```

or something like:

```
$ pulumi query -s production findPublicBuckets.ts
```

and the queries I am going to write are just going to elide the stackName, right?

I wonder if we should swap the order of these arguments?  Or add an arguments bag and move the stackName parameter into it?